src/main/java/com/binance/api/client/impl/BinanceApiWebSocketListener.java

```diff
@@ -40,8 +42,15 @@ public void onMessage(WebSocket webSocket, String text) {
     }
   }
 
+  @Override
+  public void onClosing(final WebSocket webSocket, final int code, final String reason) {
+    closing = true;
+  }
+
   @Override
   public void onFailure(WebSocket webSocket, Throwable t, Response response) {
-    throw new BinanceApiException(t);
+    if (!closing) {
```

Comment:
I wonder if there should still be some sort of notification to inform about the closure, because a regular API user might not see the closure of the socket, leaving him wondering why his application no longer works after a time.

Reply:
Unfortunately, without this, when the user calls `close` on the web socket, they user would receive a `EOFException` on this onFailure callback.  This is not intuitive IMHO.

The flag stops the `EOFException` that occurs in response to the user's `close` request.

Reply:
Yea but if the user does not close it and does not check onFailure, then the socket closes unnoticed.